docs/design/document-type.md

```diff
@@ -14,346 +14,94 @@ such as `struct`, `union`, etc. which are handled by the [Kotlin Smithy SDK](kot
 
 # Design
 
-The `document` type is an untyped JSON-like value that can take on the following types: null, boolean, string, byte,
-short, integer, long, float, double, an array of these types, or a map of these types where the key is string.
+The `document` type is an untyped JSON-like value that can take on the following types:
+* null
+* boolean
+* string
+* number (byte, short, integer, long, float, double)
+* an array of these types
+* a map of these types, keyed by string
 
 This type is best represented as a sum type (sealed class is closest we can get in Kotlin). See the
 [JSON type](https://github.com/Kotlin/kotlinx.serialization/blob/master/runtime/commonMain/src/kotlinx/serialization/json/JsonElement.kt)
-from the kotlinx.serialization lib for an example on which the following is derived. We should provide our own type but
-we may be able to internally deal with serialization by going through the one from the kotlinx.serialization library.
+from the kotlinx.serialization lib for an example on which the following is derived.
 
-````kotlin
-package com.amazonaws.smithy.runtime
+# Document
+Document directly implements the subclasses that wrap all the possible values the type can hold.
```

Comment:
The wording here is confusing. `Document` doesn't implement anything. 

Maybe something like:

> The `Document` shape will be represented as a sealed class in Kotlin with each variant (subclass) representing one of the possible values the type can hold.